tests/utils/test_archive.py

```diff
@@ -155,6 +155,57 @@ def test_archive02(self):
                 fn = "{0}.zip".format(ts)
             self.assertTrue(fn in archived_entries, "'{0}' is not in the list of unpurged entires".format(fn))
 
+    def test_archive03(self):
+        """
+        If the StateFlusher has to flush the same file, it should
+        overwrite the existing one.
+        """
+        temp_files = [
+            'Prod.0.manifest.xml',
+            'Prod.0.agentsManifest',
+            'Microsoft.Azure.Extensions.CustomScript.0.xml'
+        ]
+
+        def _write_goal_state_files(temp_files):
+            for f in temp_files:
+                self._write_file(f)
+
+        timestamp = datetime.utcnow()
+
+        _write_goal_state_files(temp_files)
+        test_subject = StateFlusher(self.tmp_dir)
+        test_subject.flush(timestamp)
+
+        # re-write all of the same files, and flush again. .flush()
+        # should overwrite the existing ones
+
```

Comment:
I don't see the test actually making sure the contents of the file, after the second flush, are the correct contents. In other words, if file X held `123` as its contents, was flushed, then was changed to hold `456` and was flushed again, the test should confirm that X contains `456` in the archive directory.

Reply:
Fixed.